src/main/java/meep/task/Storage.java

```diff
@@ -0,0 +1,128 @@
+package meep.task;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import meep.MeepException;
+import meep.ui.Ui;
+
+/**
+ * The {@code Storage} class handles the loading and saving of tasks to and from a file.
+ * It manages the storage of task data, ensuring that tasks are persisted between sessions.
+ */
+public class Storage {
+    private final String dataPath;
+
+    /**
+     * Constructs a {@code Storage} object with the specified file path for task data.
+     *
+     * @param dataPath The file path where task data will be stored.
+     */
+    public Storage(String dataPath) {
+        this.dataPath = dataPath;
+    }
+
+    /**
+     * Processes a task string and adds the corresponding task to the specified {@code TaskList}.
+     * The task string is expected to be in the format used for saving tasks to a file.
+     *
+     * @param taskString The string representation of the task.
+     * @param taskList   The {@code TaskList} to which the task will be added.
+     */
+    private static void processTaskString(String taskString, TaskList taskList) {
+        try {
+            String[] taskParts = taskString.split("\\|");
+            String taskType = taskParts[0];
+            boolean isDone = taskParts[1].equals("1");
+            String taskDescription = taskParts[2];
+            Task task;
+            switch (taskType) {
+            case "T":
+                task = new Todo(taskDescription);
+                if (isDone) {
+                    task.markAsDone();
+                }
+                taskList.addItem(task);
+                break;
+            case "D":
+                task = new Deadline(taskDescription, taskParts[3]);
+                if (isDone) {
+                    task.markAsDone();
+                }
+                taskList.addItem(task);
+                break;
+            case "E":
+                task = new Event(taskDescription, taskParts[3], taskParts[4]);
+                if (isDone) {
+                    task.markAsDone();
+                }
+                taskList.addItem(task);
+                break;
+            default:
+                break;
+            }
```

Comment:
Great use of switch!

Reply:
Thank you for your valuable feedback on the code!